oak/server/rust/oak_runtime/src/node/http/mod.rs

```diff
@@ -0,0 +1,520 @@
+            Label::decode(label_bytes).map_err(|err| {
+                warn!("Could not parse Http label: {}", err);
+                OakStatus::ErrInternal
+            })
+        }
+        None => {
+            warn!("No HTTP label found:");
+            Err(OakStatus::ErrInternal)
+        }
+    }
+}
+
+struct HttpResponseIterator {
+    runtime: RuntimeProxy,
+    response_reader: oak_abi::Handle,
+}
+
+impl HttpResponseIterator {
+    fn read_response(&self) -> Option<HttpResponse> {
+        let read_status = self
+            .runtime
+            .wait_on_channels(&[self.response_reader])
+            .map_err(|error| {
+                error!("Couldn't wait on the HTTP response channel: {:?}", error);
+            })
+            .ok()?;
+
+        if read_status[0] == ChannelReadStatus::ReadReady {
+            match self.runtime.channel_read(self.response_reader) {
+                Ok(Some(msg)) => match HttpResponse::decode(msg.data.as_slice()) {
+                    Ok(http_rsp) => {
+                        trace!(
+                            "Return response of size {}, status={:?}",
+                            http_rsp.body.len(),
+                            http_rsp.status,
+                        );
+                        Some(http_rsp)
+                    }
+                    Err(err) => {
+                        error!("Couldn't parse the HttpResponse message: {}", err);
+                        None
+                    }
+                },
+                Ok(None) => {
+                    error!("No message available on HTTP response channel");
+                    None
+                }
+                Err(status) => {
+                    error!("Couldn't read from the HTTP response channel: {:?}", status);
+                    None
+                }
+            }
+        } else if read_status[0] == ChannelReadStatus::Orphaned {
+            debug!("HTTP response channel closed");
+            None
+        } else {
+            error!(
+                "Couldn't read from the HTTP response channel: {:?}",
+                read_status[0]
+            );
+            None
+        }
+    }
+
+    fn to_response(&self) -> Response<Body> {
+        info!(
+            "Generating response for runtime {} and reader {:?}.",
+            self.runtime.node_id.0, self.response_reader
+        );
+        let mut response = Response::new(Body::empty());
+        if let Some(http_response) = self.read_response() {
+            let status_code = http_response.status as u16;
+            *response.body_mut() = Body::from(http_response.body);
+            *response.status_mut() = StatusCode::from_u16(status_code)
+                .unwrap_or_else(|_| panic!("Error when creating status code {}", status_code));
+        } else {
+            *response.status_mut() = StatusCode::from_u16(503)
+                .expect("Error when creating internal error (503) status code.");
```

Comment:
https://docs.rs/http/0.2.1/http/status/struct.StatusCode.html#associatedconstant.SERVICE_UNAVAILABLE

but also I am not sure why this particular code?